src/java/org/apache/cassandra/io/util/FileUtils.java

```diff
@@ -360,30 +325,7 @@ public static boolean isContained(File folder, File file)
 
     public static void clean(ByteBuffer buffer)
     {
-        if (buffer == null || !buffer.isDirect())
-            return;
-
-        // TODO Once we can get rid of Java 8, it's simpler to call sun.misc.Unsafe.invokeCleaner(ByteBuffer),
-        // but need to take care of the attachment handling (i.e. whether 'buf' is a duplicate or slice) - that
-        // is different in sun.misc.Unsafe.invokeCleaner and this implementation.
-
-        try
-        {
-            Object cleaner = mhDirectBufferCleaner.bindTo(buffer).invoke();
-            if (cleaner != null)
-            {
-                // ((DirectBuffer) buf).cleaner().clean();
-                mhCleanerClean.bindTo(cleaner).invoke();
-            }
-        }
-        catch (RuntimeException e)
-        {
-            throw e;
-        }
-        catch (Throwable e)
-        {
-            throw new RuntimeException(e);
-        }
+        MemoryUtil.clean(buffer);
```

Comment:
We can inline this to call MemoryUtil, but there are 42 usages (or alternative move theUnsafe into here).

Reply:
Can you refactor the 42 usages to use `MemoryUtil`? I think it makes more sense since `ByteBuffer` is not inherently file related. Not very valuable for cleaning `ByteBuffers` but removing indirection also makes inlining work better because the budget is calculated based off compiled byte code not post optimization/compilation after inlining.

Reply:
Good point - done.